service-base/src/main/java/org/eclipse/hono/service/http/HttpUtils.java

```diff
@@ -285,6 +285,33 @@ public static Integer getTimeTillDisconnect(final RoutingContext ctx) {
         return null;
     }
 
+    /**
+     * Gets the value of the {@link org.eclipse.hono.util.Constants#HEADER_TIME_TO_LIVE} HTTP header for a request.
+     * If no such header can be found, the query is searched for containing a query parameter with the same key.
+     *
+     * @param ctx The routing context containing the HTTP request.
+     * @return The time to live or {@code null} if
+     * <ul>
+     *     <li>the request doesn't contain a {@link org.eclipse.hono.util.Constants#HEADER_TIME_TO_LIVE} header 
+     *     or query parameter.</li>
+     *     <li>the contained value cannot be parsed as an Integer</li>
+     * </ul>
+     * @throws NullPointerException if context is {@code null}.
+     */
+    public static Long getTimeToLive(final RoutingContext ctx) {
```

Comment:
I think it would make sense to use `Duration` here as well.

Reply:
@ctron I have updated and pushed the changes. Please take a look.